### PR TITLE
Feature/issue 264

### PR DESCRIPTION
**Pull request by dzoccarato**
### What does it fix?
closes https://github.com/code4romania/monitorizare-vot/issues/264

### Main changes
- add extension method to get `UserType` from his claims
- add extension method to be able to build paginated result with a generic `IQueryable<T>`
- implements get notification `api/v1/notification/get/all` filtering by `UserType` from retrieved from his claim

### How has it been tested?
Manually tested:
1. added 3 different used in database:
image
_test1_ and _test3_ has same `IdNgo`
2. added some notificaion:
image
id: 2, 4, 6 are sent by users that has IdNgo = 1
3. login (from swagger)  with _test1_  user
4. call (from swagger) `api/v1/notification/get/all` from swagger, this is the result:
```json
{
  "data": [
    {
      "id": 2,
      "title": "a",
      "channel": "b",
      "body": "bb",
      "insertedAt": "2020-10-25T00:00:00",
      "senderId": 1,
      "senderIdNgo": 1,
      "senderAccount": "test1",
      "sentObserverIds": []
    },
    {
      "id": 5,
      "title": "aaa",
      "channel": "bbb",
      "body": "ccc",
      "insertedAt": "2020-10-05T00:00:00",
      "senderId": 3,
      "senderIdNgo": 1,
      "senderAccount": "test3",
      "sentObserverIds": []
    },
    {
      "id": 6,
      "title": "aaaa",
      "channel": "bbbb",
      "body": "cccc",
      "insertedAt": "2020-10-05T00:00:00",
      "senderId": 1,
      "senderIdNgo": 1,
      "senderAccount": "test1",
      "sentObserverIds": []
    }
  ],
  "totalItems": 3,
  "totalPages": 1,
  "page": 1,
  "pageSize": 20
}
```

### Question:
Is the notification model valid for you ?
https://github.com/code4romania/monitorizare-vot/commit/83f6bee3606febd0f945ecaad0b66c0a25c5463e#diff-2a94f016f852896eff12a13eb7c68faee678f21d72dd35c99ab84bf77a06d6fd

### CODE REVIEW

**Comment by aniri** [comment]
hey @dzoccarato , this looks great! could we also include the SenderNgoName next to the SenderNgoId?

Also, there's another small adjustment to be made: ngo admins of type `organizer` should be able to view all notifications. And all the other ngo admins should view the notifications sent to observers linked to the same ngo (as is implemented now).

Thanks!!

File: src/api/VoteMonitor.Api.Core/Handlers/IPaginatedQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace VoteMonitor.Api.Core.Handlers
+{
+    public interface IPaginatedQueryHandler
```
Comment (idormenco) [inline src/api/VoteMonitor.Api.Core/Handlers/IPaginatedQueryHandler.cs:8]: i don't think we need this interface at all

Comment (dzoccarato) [reply src/api/VoteMonitor.Api.Core/Handlers/IPaginatedQueryHandler.cs]: Hi @idormenco , thanks for your feedback: I've removed it

**Comment by dzoccarato** [comment]
> hey @dzoccarato , this looks great! could we also include the SenderNgoName next to the SenderNgoId?
> 
> Also, there's another small adjustment to be made: ngo admins of type `organizer` should be able to view all notifications. And all the other ngo admins should view the notifications sent to observers linked to the same ngo (as is implemented now).
> 
> Thanks!!

Hi @aniri ! Thanks for our feedback
I've added `SenderNgoName` to notification result and fix the notification get/all behavior: I misunderstood the difference between `NgoAdmin` and Organizer 😅 basically and `Organizer` is an `NgoAdmin` with some more privileges, it's correct? Should works now!

The test I've done:
- Prerequisites: database configuration: I've make some adjustment to my db data, in particular
  - NgoAdmin table:
    image
  - Nsog table:
    image
  - Notifications table:
    image

- runs request `api/v1/notification/get/all`:
  - as **test1**, all notifications:
```json
{
  "data": [
    {
      "id": 2,
      "title": "a",
      "channel": "b",
      "body": "bb",
      "insertedAt": "2020-10-25T00:00:00",
      "senderId": 1,
      "senderIdNgo": 1,
      "senderNgoName": "Code4Romania",
      "senderAccount": "test1",
      "sentObserverIds": []
    },
    {
      "id": 4,
      "title": "aa",
      "channel": "bb",
      "body": "cc",
      "insertedAt": "2020-10-05T00:00:00",
      "senderId": 2,
      "senderIdNgo": 2,
      "senderNgoName": "Guest NGO",
      "senderAccount": "test2",
      "sentObserverIds": []
    },
    {
      "id": 5,
      "title": "aaa",
      "channel": "bbb",
      "body": "ccc",
      "insertedAt": "2020-10-05T00:00:00",
      "senderId": 3,
      "senderIdNgo": 2,
      "senderNgoName": "Guest NGO",
      "senderAccount": "test3",
      "sentObserverIds": []
    },
    {
      "id": 6,
      "title": "aaaa",
      "channel": "bbbb",
      "body": "cccc",
      "insertedAt": "2020-10-05T00:00:00",
      "senderId": 1,
      "senderIdNgo": 1,
      "senderNgoName": "Code4Romania",
      "senderAccount": "test1",
      "sentObserverIds": []
    }
  ],
  "totalItems": 4,
  "totalPages": 1,
  "page": 1,
  "pageSize": 20
}
```
  - as **test2**, notification of _test2_ and _test3_ users - both has `IdNgo` = 2:
```json
{
  "data": [
    {
      "id": 4,
      "title": "aa",
      "channel": "bb",
      "body": "cc",
      "insertedAt": "2020-10-05T00:00:00",
      "senderId": 2,
      "senderIdNgo": 2,
      "senderNgoName": "Guest NGO",
      "senderAccount": "test2",
      "sentObserverIds": []
    },
    {
      "id": 5,
      "title": "aaa",
      "channel": "bbb",
      "body": "ccc",
      "insertedAt": "2020-10-05T00:00:00",
      "senderId": 3,
      "senderIdNgo": 2,
      "senderNgoName": "Guest NGO",
      "senderAccount": "test3",
      "sentObserverIds": []
    }
  ],
  "totalItems": 2,
  "totalPages": 1,
  "page": 1,
  "pageSize": 20
}
```

**Comment by aniri** [comment]
> basically and `Organizer` is an `NgoAdmin` with some more privileges, it's correct?

yep, that's right! it works perfectly now! thanks @dzoccarato

File: src/api/VoteMonitor.Api.Core/Handlers/PaginatedQueryHandlerBase.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Linq;
+
+namespace VoteMonitor.Api.Core.Handlers
+{
+    public class PaginatedQueryHandlerBase
```
Comment (idormenco) [inline src/api/VoteMonitor.Api.Core/Handlers/PaginatedQueryHandlerBase.cs:5]: Can we please move this to `VoteMonitor.Api.Core.Extensions` and rename `PaginatedQueryHandlerBase` to `QueryExtension`. 
This is my final comment i sear ! :)

**Review by idormenco — CHANGES_REQUESTED** [review]

File: src/api/VoteMonitor.Api.Core/Handlers/PaginatedQueryHandlerBase.cs (hunk 2 above)
Comment (dzoccarato) [reply src/api/VoteMonitor.Api.Core/Handlers/PaginatedQueryHandlerBase.cs]: @idormenco , done, please, take a look! :)

**Review by idormenco — APPROVED** [review]

**Pull request merged**